### PR TITLE
fix: do not render devenv.yaml for non-services

### PR DESCRIPTION
**Pull request by chatchai-outreach**
## What this PR does / why we need it

devenv.yaml should not render for non-services (libraries and CLIs)

## Jira ID

https://outreach-io.atlassian.net/browse/DT-2176

### CODE REVIEW

File: templates/devenv.yaml.tpl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- $_ := stencil.ApplyTemplate "skipIfNotService" -}}
```
Comment (jaredallard) [inline templates/devenv.yaml.tpl:1]: I think we might've actually had a reason for this... do you recall @george-e-shaw-iv?

Comment (george-e-shaw-iv) [reply templates/devenv.yaml.tpl:1]: We wanted to be able to allow libraries to run e2e tests

Comment (chatchai-outreach) [reply templates/devenv.yaml.tpl:1]: so we no longer need https://outreach-io.atlassian.net/browse/DT-2176?

Comment (george-e-shaw-iv) [reply templates/devenv.yaml.tpl:1]: Thats up for y'all to decide

**Comment by chatchai-outreach** [comment]
per discussed with @jaredallard, we no longer it

**Pull request closed without merging**